pkg/trace/api/api_test.go

```diff
@@ -368,6 +368,31 @@ func TestReceiverDecodingError(t *testing.T) {
 	})
 }
 
+func TestReceiverUnexpectedEOF(t *testing.T) {
+	assert := assert.New(t)
+	conf := newTestReceiverConfig()
+	r := newTestReceiverFromConfig(conf)
+	server := httptest.NewServer(http.HandlerFunc(r.handleWithVersion(v05, r.handleTraces)))
```

Comment:
This is fine, but I'm curious - why not `r.Start` to use what actually runs?

Reply:
This was copied and pasted from elsewhere in this file, so I guess you could ask the same question about `TestReceiverDecodingError`

<img width="1006" alt="Screenshot 2020-09-07 at 13 16 08" src="https://user-images.githubusercontent.com/16439049/92386775-5bc8a880-f10c-11ea-89bb-3b4526ffdfa9.png">

Reply:
🤷‍♂️ Indeed. I think `r.Start` will be better because we're testing the actual thing that the agent runs, without leaving anything out. No idea why I did that then or what the history of that code is.